Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+                'ResourceType': 'volume',
+                'Tags': parse_tag_field(args.get('tags'))}]
+        })
+
+    response = client.create_volume(**kwargs)
+
+    data = ({
+        'AvailabilityZone': response['AvailabilityZone'],
+        'CreateTime': datetime.strftime(response['CreateTime'], '%Y-%m-%dT%H:%M:%SZ'),
+        'Encrypted': response['Encrypted'],
+        'Size': response['Size'],
+        'State': response['State'],
+        'VolumeId': response['VolumeId'],
+        'Iops': response['Iops'],
+        'VolumeType': response['VolumeType'],
+        'Region': obj['_user_provided_options']['region_name'],
+    })
+    if 'SnapshotId' in response:
+        data.update({'SnapshotId': response['SnapshotId']})
+    if 'KmsKeyId' in response:
+        data.update({'KmsKeyId': response['KmsKeyId']})
+    if 'Tags' in response:
+        for tag in response['Tags']:
+            data.update({
+                tag['Key']: tag['Value']
+            })
+
+    ec = {'AWS.EC2.Volumes': data}
+    human_readable = tableToMarkdown('AWS EC2 Volumes', data)
+    return_outputs(human_readable, ec)
+
+
+def attach_volume_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    kwargs = {
+        'Device': args.get('device'),
+        'InstanceId': args.get('instanceId'),
+        'VolumeId': args.get('volumeId'),
+    }
+    response = client.attach_volume(**kwargs)
+    data = ({
+        'AttachTime': datetime.strftime(response['AttachTime'], '%Y-%m-%dT%H:%M:%SZ'),
+        'Device': response['Device'],
+        'InstanceId': response['InstanceId'],
+        'State': response['State'],
+        'VolumeId': response['VolumeId'],
+    })
+    if 'DeleteOnTermination' in response:
+        data.update({'DeleteOnTermination': response['DeleteOnTermination']})
+
+    ec = {'AWS.EC2.Volumes(val.VolumeId === obj.VolumeId).Attachments': data}
+    human_readable = tableToMarkdown('AWS EC2 Volume Attachments', data)
+    return_outputs(human_readable, ec)
+
+
+def detach_volume_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    kwargs = {'VolumeId': args.get('volumeId')}
+
+    if args.get('force') is not None:
+        kwargs.update({'Force': True if args.get('force') == 'True' else False})
+    if args.get('device') is not None:
+        kwargs.update({'Device': int(args.get('device'))})
+    if args.get('instanceId') is not None:
+        kwargs.update({'InstanceId': args.get('instanceId')})
+
+    response = client.detach_volume(**kwargs)
+    data = ({
+        'AttachTime': datetime.strftime(response['AttachTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.